app/src/main/kotlin/xyz/duckee/android/MainActivity.kt

```diff
@@ -109,20 +107,47 @@ class MainActivity : ComponentActivity() {
                 }
             }
         }
+
+        // Handle deeplink when the app is opened from a deeplink
+        handleDeeplink(intent)
     }
 
     override fun onResume() {
         super.onResume()
 
-        Firebase.dynamicLinks
-            .getDynamicLink(intent)
-            .addOnSuccessListener(this) { pendingDynamicLinkData: PendingDynamicLinkData? ->
-                pendingDynamicLinkData?.link?.let { deepLink ->
-                    // Handle Deeplink with Jetpack NavigationComponent
-                    navigationController?.navigate(deepLink)
-                }
+        // Handle deferred deeplink
+        Airbridge.handleDeferredDeeplink { uri ->
+            uri?.let {
+                // Handle the deferred deeplink URI
+                // For example, navigate to a specific screen based on the URI
+                handleDeeplinkUri(it)
+            }
+        }
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        intent?.let { handleDeeplink(it) }
+    }
+
+    private fun handleDeeplink(intent: Intent) {
+        Airbridge.handleDeeplink(intent) { uri ->
+            // Handle the deeplink URI
+            // For example, navigate to a specific screen based on the URI
+            handleDeeplinkUri(uri)
+        }
+    }
+
+    private fun handleDeeplinkUri(uri: Uri) {
+        // Implement your deeplink handling logic here
+        // For example, navigate to specific screens based on the URI
+        when {
+            uri.path?.startsWith("/product/") == true -> {
+                val productId = uri.lastPathSegment
+                // Navigate to product detail screen
+                navigationController?.navigate("product/$productId")
             }
-            .addOnFailureListener(this) { e ->
-                Timber.tag("[DuckeeMainActivity]").w(e, "getDynamicLink:onFailure") }
+            // Add more cases as needed
+        }
     }
```

Comment:
Added `handleDeeplinkUri()` method to implement custom deeplink handling logic. This method should be customized based on your app's specific deeplink structure and navigation requirements. The example provided demonstrates how to handle a product detail deeplink.